Assets/MRTK/Providers/LeapMotion/LeapMotionDeviceManager.cs

```diff
@@ -0,0 +1,289 @@
+using System.Collections.Generic;
+using Unity.Profiling;
+using System;
+
+#if LEAPMOTIONCORE_PRESENT
+using Leap;
+using Leap.Unity;
+using Leap.Unity.Attachments;
+#endif
+
+namespace Microsoft.MixedReality.Toolkit.LeapMotion.Input
+{
+    [MixedRealityDataProvider(
+        typeof(IMixedRealityInputSystem),
+        SupportedPlatforms.WindowsStandalone | SupportedPlatforms.WindowsEditor,
+        "Leap Motion Device Manager",
+        "LeapMotion/Profiles/LeapMotionDeviceManagerProfile.asset",
+        "MixedRealityToolkit.Providers",
+        true)]
+    /// <summary>
+    /// Class that detects the tracking state of leap motion hands.  This class will only run if the Leap Motion Core Assets are in the project and the Leap Motion Device
+    /// Manager data provider has been added in the input system configuration profile.
+    /// </summary>
+    public class LeapMotionDeviceManager : BaseInputDeviceManager, IMixedRealityCapabilityCheck
+    {
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        /// <param name="inputSystem">The <see cref="Microsoft.MixedReality.Toolkit.Input.IMixedRealityInputSystem"/> instance that receives data from this provider.</param>
+        /// <param name="name">Friendly name of the service.</param>
+        /// <param name="priority">Service priority. Used to determine order of instantiation.</param>
+        /// <param name="profile">The service's configuration profile.</param>
+        public LeapMotionDeviceManager(
+            IMixedRealityInputSystem inputSystem,
+            string name = null,
+            uint priority = DefaultPriority,
+            BaseMixedRealityProfile profile = null) : base(inputSystem, name, priority, profile) { }
+
+
+#region IMixedRealityCapabilityCheck Implementation
+
+        /// <inheritdoc />
+        public bool CheckCapability(MixedRealityCapability capability)
+        {
+            // Leap Motion only supports Articulated Hands
+            return (capability == MixedRealityCapability.ArticulatedHand);
+        }
+
+
+#endregion IMixedRealityCapabilityCheck Implementation
+#if LEAPMOTIONCORE_PRESENT
+
+        /// <summary>
+        /// The profile that contains settings for the Leap Motion Device Manager input data provider.  This profile is nested under 
+        /// Input > Input Data Providers > Leap Motion Device Manager in the MixedRealityToolkit object in the hierarchy.
+        /// </summary>
+        public LeapMotionDeviceManagerProfile SettingsProfile => ConfigurationProfile as LeapMotionDeviceManagerProfile;
+
+        /// <summary>
+        /// If true, the leap motion controller is connected and detected.
+        /// </summary>
+        public bool IsLeapConnected => LeapMotionServiceProvider.IsConnected();
+
+        /// <summary>
+        /// The LeapServiceProvider is added to the scene at runtime in OnEnable. 
+        /// </summary>
+        public LeapServiceProvider LeapMotionServiceProvider { get; protected set; }
```

Comment:
It might be good to do a quick pass on which of these `public` properties need to remain public and which can be made private. I think some of these are now passed directly to the class that needs them (like this one and the hand class, I believe). I think it's definitely better to use the minimum expected / needed scope up front and expand if necessary.

Reply:
Ok that makes sense,  I changed everything to private except the LeapServiceProvider.  I think the LeapServiceProvider should remain public as the minimum.

Reply:
Sounds good to me! You definitely know best